rules.install

```diff
@@ -4,6 +4,45 @@
  * Rules - Installation file.
  */
 
+/**
+ * Implements hook_requirements().
+ */
+function rules_requirements($phase) {
+  $requirements = array();
+  if ($phase == 'update') {
+    // Required Backdrop modules that were not part of D7 installation.
+    $backdrop_modules = array(
+      'entity_ui' => 'Entity UI',
+      'entity_plus' => 'Entity Plus',
+    );
+    foreach ($backdrop_modules as $module => $module_name) {
+      if (module_exists($module)) {
+        // The module is already installed and enabled, nothing more to do.
+        continue;
+      }
+      $module_files = backdrop_system_listing('/' . $module . '\.module$/', 'modules', 'name', 0);;
+      if (empty($module_files)) {
+        // Module doesn't exist in the code base, so halt the upgrade process.
+        $requirements[] = array(
+          'title' => $module_name,
+          'description' => t('Please add the Backdrop module @module_name to your installation; it is required for the Rules module.', array('@module_name' => $module_name)),
+          'severity' => REQUIREMENT_ERROR,
+        );
+      }
```

Comment:
The if statement that ends here is not necessary. The core update.php script already  performs this check to see if all required modules are present in the modules folder, and if not, it stops the process completely with an "Unresolved dependency" error before any update.